nyx_client/nyx_client/client.py

```diff
@@ -551,27 +554,30 @@ def create_data(
         name: str,
         title: str,
         description: str,
-        size: int,
         genre: str,
         categories: Sequence[str],
-        download_url: str,
         content_type: str,
         lang: str = "en",
         status: str = "published",
         preview: str = "",
+        size: int | None = None,
         price: int | None = None,
         license_url: str | None = None,
+        download_url: str | None = None,
+        file: TextIOBase | None = None,
     ) -> Data:
         """Create new data in the system.
 
         Args:
             name: The unique identifier for the data.
             title: The display title of the data.
             description: A detailed description of the data.
-            size: The size of the data, typically in bytes.
+            size: Approximate size of the data, if a file is provided the size
+                  will be calculated
             genre: The genre or category of the data.
             categories: A list of categories the data belongs to.
             download_url: The URL where the data can be downloaded.
+            file: the file like object (TextIOBase) that you wish to upload
```

Comment:
Just leaving note here: Maybe we should expect `RawIOBase` unless we decide to not address the text encoding in the server - in which case this would mean we're guaranteed utf8.